### PR TITLE
Upgrading from Buidler to Hardhat

### PR DESCRIPTION
**Pull request by aodhgan**

### CODE REVIEW

File: hardhat.networks.js [hunk 1]
```diff
@@ -19,12 +14,12 @@ const networks = {
   }
 }
 
-if (process.env.USE_BUIDLER_EVM_ACCOUNTS) {
-  networks.buidlerevm.accounts = process.env.USE_BUIDLER_EVM_ACCOUNTS.split(/\s+/).map(privateKey => ({
-    privateKey,
-    balance: ethers.utils.parseEther('10000000').toHexString()
-  }))
-}
+// if (process.env.USE_BUIDLER_EVM_ACCOUNTS) {
```
Comment (aodhgan) [inline hardhat.networks.js:17]: @asselstine Do we still need this?

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-v10.20.1
+v14.15.3
```
Comment (asselstine) [inline .nvmrc:1]: 🤩

File: hardhat.networks.js (hunk 1 above)
Comment (asselstine) [reply hardhat.networks.js:17]: No we don't!

File: js/deployTestPool.js [hunk 3]
```diff
@@ -25,6 +24,10 @@ async function deployTestPool({
   overrides = { gasLimit: 20000000 }
 }) {
   await deployments.fixture()
+  // const ERC20Mintable = require('../build/ERC20Mintable.json')
+  // const ERC20Mintable = await hre.artifacts.readArtifact("ERC20Mintable")
```
Comment (asselstine) [inline js/deployTestPool.js:28]: ✂️

File: js/deployTestPool.js [hunk 4]
```diff
@@ -38,7 +41,9 @@ async function deployTestPool({
     debug('Deploying test stake pool')
   }
 
-  let governanceToken = await deployContract(wallet, ERC20Mintable, ['Governance Token', 'GOV'], overrides)
+
+  // let governanceToken = await deployContract(wallet, ERC20Mintable, [], overrides)
```
Comment (asselstine) [inline js/deployTestPool.js:45]: ✂️

File: js/deployTestPool.js [hunk 5]
```diff
@@ -48,15 +53,17 @@ async function deployTestPool({
...
+  const token = await hardhat.ethers.getContractAt('ERC20Mintable', tokenResult.address, wallet)
+  const cToken = await hardhat.ethers.getContractAt('CTokenMock', cTokenResult.address, wallet)
+  const yToken = await hardhat.ethers.getContractAt('yVaultMock', yTokenResult.address, wallet)
+  const comptroller = await hardhat.ethers.getContractAt('ComptrollerHarness', comptrollerResult.address, wallet)
+  const poolBuilder = await hardhat.ethers.getContractAt('PoolWithMultipleWinnersBuilder', poolWithMultipleWinnersBuilderResult.address, wallet)
+
 
-  let linkToken = await deployContract(wallet, ERC20Mintable, ['Link Token', 'LINK'], overrides)
-  let rngServiceMock = await buidler.ethers.getContractAt('RNGServiceMock', rngServiceMockResult.address, wallet)
+  // let linkToken = await deployContract(wallet, ERC20Mintable, ['Link Token', 'LINK'], overrides)
```
Comment (asselstine) [inline js/deployTestPool.js:64]: ✂️

File: package.json [hunk 6]
```diff
@@ -11,46 +11,47 @@
     "reinstall": "rm -rf node_modules/ && rm -f yarn.lock && yarn",
     "verify": "yarn hint && yarn test",
     "hint": "solhint \"contracts/**/*.sol\"",
-    "test": "OPTIMIZER_DISABLED=true buidler test",
-    "compile": "buidler --show-stack-traces --max-memory 8192 compile",
-    "start": "echo 'Starting node for local testing....\n'; rm -rf deployments/localhost_31337; DISABLE_HARNESS=true USE_BUIDLER_EVM_ACCOUNTS=$BUIDLER_EVM_ACCOUNTS buidler node",
-    "start-gas": "echo 'Starting node for gas testing....\n'; rm -rf deployments/localhost_31337; DISABLE_HARNESS=true buidler node",
+    "test": "hardhat test",
+    "compile": "hardhat --show-stack-traces --max-memory 8192 compile",
+    "start": "echo 'Starting node for local testing....\n'; rm -rf deployments/localhost_31337; DISABLE_HARNESS=true USE_BUIDLER_EVM_ACCOUNTS=$BUIDLER_EVM_ACCOUNTS hardhat node",
```
Comment (asselstine) [inline package.json:16]: Let's strip out USE_BUIDLER_EVM_ACCOUNTS env var here

File: test/BalanceDripManagerExposed.test.js [hunk 7]
```diff
@@ -22,18 +19,32 @@ describe('BalanceDripManagerExposed', function() {
...
 
     debug({ dripExposed: dripExposed.address })
 
-    measure = await deployContract(wallet, ERC20Mintable, ['Measure Token', 'MTKN'], overrides)
-    drip1 = await deployContract(wallet, ERC20Mintable, ['Drip Token 1', 'DRIP1'], overrides)
-    drip2 = await deployContract(wallet, ERC20Mintable, ['Drip Token 2', 'DRIP2'], overrides)
-    drip3 = await deployMockContract(wallet, IERC20.abi)
+    // measure = await deployContract(wallet, ERC20Mintable, ['Measure Token', 'MTKN'], overrides)
+     const ERC20MintableContract =  await hre.ethers.getContractFactory("ERC20Mintable", wallet, overrides)
+   measure = await ERC20MintableContract.deploy('Measure Token', 'MTKN')
```
Comment (asselstine) [inline test/BalanceDripManagerExposed.test.js:35]: whitespace

File: test/BalanceDripManagerExposed.test.js [hunk 8]
```diff
@@ -22,18 +19,32 @@ describe('BalanceDripManagerExposed', function() {
...
 
-    dripExposed = await deployContract(wallet, BalanceDripManagerExposed, [], overrides)
 
     debug({ dripExposed: dripExposed.address })
 
-    measure = await deployContract(wallet, ERC20Mintable, ['Measure Token', 'MTKN'], overrides)
-    drip1 = await deployContract(wallet, ERC20Mintable, ['Drip Token 1', 'DRIP1'], overrides)
-    drip2 = await deployContract(wallet, ERC20Mintable, ['Drip Token 2', 'DRIP2'], overrides)
-    drip3 = await deployMockContract(wallet, IERC20.abi)
+    // measure = await deployContract(wallet, ERC20Mintable, ['Measure Token', 'MTKN'], overrides)
```
Comment (asselstine) [inline test/BalanceDripManagerExposed.test.js:33]: ✂️

File: test/BalanceDripManagerExposed.test.js [hunk 9]
```diff
@@ -22,18 +19,32 @@ describe('BalanceDripManagerExposed', function() {
...
+     const ERC20MintableContract =  await hre.ethers.getContractFactory("ERC20Mintable", wallet, overrides)
+   measure = await ERC20MintableContract.deploy('Measure Token', 'MTKN')
+
+    // drip1 = await deployContract(wallet, ERC20Mintable, ['Drip Token 1', 'DRIP1'], overrides)
+    drip1 = await ERC20MintableContract.deploy('Drip Token 1', 'DRIP1')
+
+    // drip2 = await deployContract(wallet, ERC20Mintable, ['Drip Token 2', 'DRIP2'], overrides)
+    drip2 = await ERC20MintableContract.deploy('Drip Token 2', 'DRIP2')
+    
+    // drip3 = await deployMockContract(wallet, IERC20.abi)
```
Comment (asselstine) [inline test/BalanceDripManagerExposed.test.js:43]: ✂️  for all commented out code

File: test/CTokenMock.test.js [hunk 10]
```diff
@@ -15,47 +12,50 @@ describe('CTokenMock contract', function() {
...
-        ])
+        [wallet, otherWallet] = await hardhat.ethers.getSigners()
+        const ERC20Mintable = await hre.ethers.getContractFactory("ERC20Mintable", wallet)
+        token = await ERC20Mintable.deploy('Test Token', 'TEST')
+        // token = await deployContract(wallet, ERC20Mintable, ['Test Token', 'TEST'])
+        const CTokenMockContract = await hre.ethers.getContractFactory("CTokenMock", wallet)
+        cTokenMock = await CTokenMockContract.deploy(token.address, ethers.utils.parseEther('0.01'))
+        // cTokenMock = await deployContract(wallet, CTokenMock, [
+        //     token.address, ethers.utils.parseEther('0.01')
+        // ])
```
Comment (asselstine) [inline test/CTokenMock.test.js:23]: ✂️ for all commented out code

File: test/CompoundPrizePoolProxyFactory.test.js [hunk 11]
```diff
@@ -12,10 +10,11 @@ describe('CompoundPrizePoolProxyFactory', () => {
...
   beforeEach(async () => {
-    [wallet, wallet2] = await buidler.ethers.getSigners()
-    provider = buidler.ethers.provider
-
-    factory = await deployContract(wallet, CompoundPrizePoolProxyFactory, [], overrides)
+    [wallet, wallet2] = await hardhat.ethers.getSigners()
+    provider = hardhat.ethers.provider
+    const CompoundPrizePoolProxyFactory =  await hre.ethers.getContractFactory("CompoundPrizePoolProxyFactory", wallet, overrides)
+    factory = await CompoundPrizePoolProxyFactory.deploy()
+    // factory = await deployContract(wallet, CompoundPrizePoolProxyFactory, [], overrides)
```
Comment (asselstine) [inline test/CompoundPrizePoolProxyFactory.test.js:17]: ✂️

File: test/VolumeDripManagerExposed.test.js [hunk 12]
```diff
@@ -27,16 +25,25 @@ describe('VolumeDripManagerExposed', function() {
...
-    drip1 = await deployContract(wallet, ERC20Mintable, ['Drip Token 1', 'DRIP1'], overrides)
-    drip2 = await deployContract(wallet, ERC20Mintable, ['Drip Token 2', 'DRIP2'], overrides)
-    drip3 = await deployMockContract(wallet, IERC20.abi)
+    const IERC20 = await hre.ethers.getContractFactory("ERC20Mintable", wallet, overrides)
+    // measure = await deployContract(wallet, ERC20Mintable, ['Measure Token', 'MTKN'], overrides)
+    measure = await IERC20.deploy('Measure Token', 'MTKN')
+    // drip1 = await deployContract(wallet, ERC20Mintable, ['Drip Token 1', 'DRIP1'], overrides)
+    drip1 = await IERC20.deploy('Drip Token 1', 'DRIP1')
+
+    // drip2 = await deployContract(wallet, ERC20Mintable, ['Drip Token 2', 'DRIP2'], overrides)
```
Comment (asselstine) [inline test/VolumeDripManagerExposed.test.js:42]: ✂️  all commented out code

File: test/features/support/PoolEnv.js [hunk 13]
```diff
@@ -59,15 +57,21 @@ function PoolEnv() {
...
     const externalAwardAddresses = []
     this.externalERC20Awards = {}
+
+    const ERC20Mintable = await hre.ethers.getContractFactory("ERC20Mintable", this.wallets[0], this.overrides)
+    const ERC721Mintable = await hre.ethers.getContractFactory("ERC721Mintable", this.wallets[0], this.overrides)
+
+
     for (var i = 0; i < externalERC20Awards.length; i++) {
-      this.externalERC20Awards[externalERC20Awards[i]] = await deployContract(this.wallets[0], ERC20Mintable, [`External ERC20 Token ${i+1}`, `ETKN${i+1}`])
+      // this.externalERC20Awards[externalERC20Awards[i]] = await deployContract(this.wallets[0], ERC20Mintable, [`External ERC20 Token ${i+1}`, `ETKN${i+1}`])
```
Comment (asselstine) [inline test/features/support/PoolEnv.js:66]: ✂️

File: test/features/support/PoolEnv.js [hunk 14]
```diff
@@ -59,15 +57,21 @@ function PoolEnv() {
...
+      this.externalERC20Awards[externalERC20Awards[i]] = await ERC20Mintable.deploy(`External ERC20 Token ${i+1}`, `ETKN${i+1}`)
       const address = this.externalERC20Awards[externalERC20Awards[i]].address;
       await this.env.prizeStrategy.addExternalErc20Award(address)
       externalAwardAddresses.push(address)
     }
 
-    this.externalErc721Award = await deployContract(this.wallets[0], ERC721Mintable, [])
-
+    // this.externalErc721Award = await deployContract(this.wallets[0], ERC721Mintable, [])
```
Comment (asselstine) [inline test/features/support/PoolEnv.js:73]: ✂️

File: test/features/support/PoolEnv.js [hunk 15]
```diff
@@ -386,8 +390,8 @@ function PoolEnv() {
 
   //   for (let userIndex = 1; userIndex < 3; userIndex++) {
   //     let wallet = await this.wallet(userIndex)
-  //     let chance = await this.env.ticket.chanceOf(wallet._address)
-  //     console.log(chalk.green(`User ${userIndex} (${wallet._address}) chances: ${ethers.utils.formatEther(chance)}`))
+  //     let chance = await this.env.ticket.chanceOf(wallet.address)
+  //     console.log(chalk.green(`User ${userIndex} (${wallet.address}) chances: ${ethers.utils.formatEther(chance)}`))
```
Comment (asselstine) [inline test/features/support/PoolEnv.js:394]: ✂️  is this fxn even needed anymore?  whats happening here?

File: test/helpers/deployMockContract.js [hunk 16]
```diff
@@ -39,10 +39,10 @@ function createMock(abi, mockContractInstance) {
...
 
 async function deployMockContract(wallet, abi) {
-  console.warn(
-    'deployMockContract is an experimental feature. ' +
-    'Breaking changes will not result in a new major version'
-  );
+  // console.warn(
+  //   'deployMockContract is an experimental feature. ' +
+  //   'Breaking changes will not result in a new major version'
+  // );
```
Comment (asselstine) [inline test/helpers/deployMockContract.js:45]: ✂️

File: test/yVaultMock.test.js [hunk 17]
```diff
@@ -1,57 +1,58 @@
...
-    erc20token = await deployContract(wallet, ERC20Mintable, ['Token', 'TOKE'], overrides)
+    const ERC20MintableContract =  await hre.ethers.getContractFactory("ERC20Mintable", wallet, overrides)
+   
+    
+    erc20token = await ERC20MintableContract.deploy("TOKEN", "TOKE")
 
     debug('creating vault...')
-    vault = await deployContract(wallet, yVaultMock, [erc20token.address], overrides)
+    const yVaultMock =  await hre.ethers.getContractFactory("yVaultMock", wallet, overrides)
+    // vault = await deployContract(wallet, yVaultMock, [erc20token.address], overrides)
```
Comment (asselstine) [inline test/yVaultMock.test.js:28]: ✂️

**Review by asselstine — CHANGES_REQUESTED** [review]

File: test/features/support/PoolEnv.js (hunk 15 above)
Comment (aodhgan) [reply test/features/support/PoolEnv.js:394]: I dont know! This was already commented out. I've removed it.

**Pull request merged**